packages/@aws-cdk/aws-cloudwatch/lib/env-tokens.ts

```diff
@@ -0,0 +1,75 @@
+import * as cdk from '@aws-cdk/core';
+import { dropUndefined } from './metric-util';
+
+/**
+ * Make a Token that renders to given region if used in a different stack, otherwise undefined
+ */
+export function regionIfDifferentFromStack(region: string): string {
+  return cdk.Token.asString(new StackDependentToken(region, stack => stack.region));
+}
+
+/**
+ * Make a Token that renders to given account if used in a different stack, otherwise undefined
+ */
+export function accountIfDifferentFromStack(account: string): string {
+  return cdk.Token.asString(new StackDependentToken(account, stack => stack.account));
+}
+
+/**
+ * A lazy token that requires an instance of Stack to evaluate
+ */
```

Comment:
Didn't finish the sentence?

Reply:
Not really.

It requires an instance of stack to evaluate [itself].